lib/charms/opensearch/v0/opensearch_nodes_exclusions.py

```diff
@@ -44,13 +44,14 @@ def __init__(self, charm):
 
         self._scope = Scope.APP if self._charm.unit.is_leader() else Scope.UNIT
 
-    def add_current(self) -> None:
+    def add_current(self, restart=False) -> None:
         """Add Voting and alloc exclusions."""
         if (self._node.is_cm_eligible() or self._node.is_voting_only()) and not self._add_voting():
             logger.error(f"Failed to add voting exclusion: {self._node.name}.")
 
-        if self._node.is_data() and not self._add_allocations():
-            logger.error(f"Failed to add shard allocation exclusion: {self._node.name}.")
+        if not restart:
```

Comment:
This can be "dangerous" as we may lose non-replicated shards.

Reply:
You're right, but I believe it may be an acceptable risk in general - from a similar section on the [elastic docs](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/rolling-upgrades.html#_upgrading_your_cluster):

> When you shut down a data node, the allocation process waits for index.unassigned.node_left.delayed_timeout (by default, one minute) before starting to replicate the shards on that node to other nodes in the cluster, **which can involve a lot of I/O. Since the node is shortly going to be restarted, this I/O is unnecessary.** You can avoid racing the clock by [disabling allocation](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/modules-cluster.html#cluster-routing-allocation-enable) of replicas before shutting down [data nodes](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/modules-node.html#data-node):